www/js/controllers.js

```diff
@@ -48,55 +47,94 @@ app.controller('HistoryController',function($rootScope, $scope, Auth, UserFinish
     $scope.authUser = Auth.$getAuth();
     $scope.finishedItems = UserFinishedTasks($scope.authUser.uid);
     $scope.clearAll = function(){
-       // finishedItemsRef.remove();
-        console.log(">>>> "+UserFinishedTasks($scope.authUser.uid).length);
+        console.log(">>>> " +UserFinishedTasks($scope.authUser.uid).length);
     }
 });
-app.controller('ToDoListActiveController', function ($rootScope, $scope, $ionicModal, Auth, UserTasks,UserFinishedTasks,$ionicListDelegate) {
-    $scope.authUser = Auth.$getAuth();
-    $scope.items = UserTasks($scope.authUser.uid);
-    $scope.finishedItems = UserFinishedTasks($scope.authUser.uid);
+app.controller('TaskController', function($scope, $state, $stateParams, $ionicHistory,
+                                          UserTask, UserTasks, Auth, $cordovaDatePicker, $interval,
+                                          $ionicListDelegate) {
 
+    $scope.authUser = Auth.$getAuth();
+    $scope.isNewTask = $stateParams.taskId == 'new';
+    $scope.username = getDisplayNameFromAuth(Auth.$getAuth()).split(" ").shift();
+    var ownerId = $stateParams.ownerId;
+    if (!ownerId) {
+        ownerId = $scope.authUser.uid;
+    }
+    $scope.setTimer = function(task){
+        if(task.dueDate == 0){
```

Comment:
you could eliminate else if you convert condition to opposite. `simply say if (task.dueDate)`. empty if blocks looks amateur.